core/blockchain.go

```diff
@@ -2546,6 +2558,7 @@ func (bc *BlockChain) ReadValidatorSnapshot(addr common.Address) (*staking.Valid
 }
 
 // WriteValidatorSnapshots writes the snapshot of provided list of validators
+// Note: this should only be called within the blockchain insertBlock process.
```

Comment:
In this case, we could make it low case method to avoid accidentally called outside.

Reply:
Good point. Done for other cases.